gr-fec/lib/cc_encoder_impl.cc

```diff
@@ -71,6 +71,10 @@ cc_encoder_impl::cc_encoder_impl(int frame_size,
             "cc_encoder: Number of polynomials must be the same as the value of rate");
     }
 
+    if (k > 32) {
```

Comment:
Since we're here with this specific argument check, can you also add checks for other variables where it makes sense? For example: k > 0 (yes?) ... rate > 0 (yes?), start_state has to be in (0,(2^k)-1) (yes?), frame_size > 0 ... any others?

Reply:
Yes, more checks can be added. I will do so this week.